homeassistant/config_entries.py

```diff
@@ -563,3 +585,64 @@ def _async_in_progress(self):
         return [flw for flw in self.hass.config_entries.flow.async_progress()
                 if flw['handler'] == self.handler and
                 flw['flow_id'] != self.flow_id]
+
+    @staticmethod
+    def async_get_options_flow():
+        """"""
+        return None
+
+
+class Options:
+    """"""
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """"""
+        self.hass = hass
+        self.flow = data_entry_flow.FlowManager(
+            hass, self._async_create_flow, self._async_finish_flow)
+        self.active_options = {}
+
+    @callback
+    def async_domains(self) -> List[str]:
```

Comment:
This is not necessary. I think the way the frontend will discover options is by looking at the options value of a config entry.

If `entry.options is None`, there are no config options, if it's a dictionary, we can start an options flow.

Reply:
So we can drop this method.

Reply:
Ok. Great! So that will cover the case when a config entry is already setup.

My current thoughts is to redirect finalising the _async_finish_flow if the config flow supports options. Or is there a better way. I'm still not seeing how I would properly inform the frontend about going from config entry to config option.